keymage.js

```diff
@@ -8,8 +8,7 @@
 (function(define, undefined) {
 define(function() {
     var VERSION = '1.1.3';
-    var isOsx = typeof navigator !== 'undefined' &&
-        ~navigator.userAgent.indexOf('Mac OS X');
+    var isOsx = /^mac/i.test(navigator.platform);
```

Comment:
>  it's deprecated

My bad: I did not check with MDN, only with [whatwg spec](https://html.spec.whatwg.org/multipage/#dom-navigator-platform).

> Plus using regexp is going to be slower than indexOf anyway.

1. All contemporary JS VMs compile regular expressions to native code. Please, consider [this benchmark](https://benchmarksgame.alioth.debian.org/u64q/performance.php?test=regexdna): despite V8 uses only one core, it is pretty high on the list. Before Rust and PHP JITs it was the first. If the machine was single-core, it would be #1.

2. About this very case: in most cases, `iOsx` will be `false` (according to number of Macs sold). `indexOf` case will go through all the (quite long) string . `/^mac/i` is bound to beginning of the string and will check only first character and abort immediately. In case of OS X, `indexOf` will loop through less characters, but still will do more work than regular expression.

3. `let opinion = true` This line of code runs only once. It does not really matter *how fast*, it is about *how readable*. `indexOf` solution intents to check whether the string `includes` given substring. Instead, it gets index of the substring and flips its bits. RegExp case with `test` is more straightforward.

4. Lets go back to performance. [`KEYS`](https://github.com/piranha/keymage/blob/master/keymage.js#L28) hashtable has invalid identifiers as keys and thus transits to dictionary mode in V8. [`for in`](https://github.com/piranha/keymage/blob/master/keymage.js#L84) called on it prevents the whole function from being optimized by Crankshaft. Furthermore, semantically [`for in`](http://www.ecma-international.org/ecma-262/6.0/#sec-runtime-semantics-forin-div-ofbodyevaluation-lhs-stmt-iterator-lhskind-labelset) is for iterating [`[[Prototype]]`](http://www.ecma-international.org/ecma-262/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-getprototypeof) chain. In V8/JSC it fails [this test case](https://github.com/tc39/test262/blob/master/test/language/statements/for-in/12.6.4-2.js) so I would rather use [`Object.keys`](http://www.ecma-international.org/ecma-262/6.0/#sec-object.keys).

Is supporting IE <= 8 without [`es5shim`](https://github.com/es-shims/es5-shim) an issue?